### ISSUE TITLE
getting <unk> with character model

### ISSUE THREAD

**Issue by anjul1008** [wenet-e2e/wenet#1550]
Hi,
I have trained a model from librispeech with 5000 vocab bpe model and it's working as expected.
But when I trained a char model with the same conf, looks model convergence well as the loss was below 2, but getting only <unk> in decoded output.

results and training logs are below for reference,
decoded o/p: 
`utt1 <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk>`
`utt2 <unk> <unk> <unk>  `
`utt3 <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> `

logs:
<img width="769" alt="Screenshot 2022-11-10 at 1 00 58 PM" src="https://user-images.githubusercontent.com/25085268/201028659-86d5be15-33c3-4e97-b59f-9b79de3f7737.png">

any suggestions on what mistake i have made?
thanks

**Comment by pengzhendong** [comment]
Plz check `exp/data/dict/lang_char.txt`.

**Comment by anjul1008** [comment]
hi,
looks fine.

```
<blank> 0
<unk> 1
<space> 2
' 2
A 3
B 4
C 5
D 6
E 7
F 8
G 9
H 10
I 11
J 12
K 13
L 14
M 15
N 16
O 17
P 18
Q 19
R 20
S 21
T 22
U 23
V 24
W 25
X 26
Y 27
Z 28
<sos/eos> 30
```

**Comment by pengzhendong** [comment]
Try to print the tokens here: https://github.com/wenet-e2e/wenet/blob/1d9b0bf066aae3170c8c084ead6260f1f76b4fe5/wenet/dataset/processor.py#L403

**Comment by anjul1008** [comment]
oh, i had added `split_with_space` to dataset_conf that splits sentence into words not char, and all words mapped to <unk>.
thanks, data conf issue is resolved now.

let it run for a couple of epochs and I'll update it here.

**Comment by anjul1008** [comment]
Hi, after one epoch terminal logs look like,
```
2022-11-10 16:47:02,001 WARNING NaN or Inf found in input tensor.
2022-11-10 16:47:04,025 WARNING NaN or Inf found in input tensor.
2022-11-10 16:47:05,200 WARNING NaN or Inf found in input tensor.
2022-11-10 16:47:06,385 WARNING NaN or Inf found in input tensor.
2022-11-10 16:47:08,350 WARNING NaN or Inf found in input tensor.
2022-11-10 16:47:08,381 DEBUG TRAIN Batch 1/1000 loss inf loss_att 165.088806 loss_ctc inf lr 0.00014175 rank 1
2022-11-10 16:47:08,382 DEBUG TRAIN Batch 1/1000 loss inf loss_att 95.901802 loss_ctc inf lr 0.00014200 rank 0
```
TensorFlow doesn't show anything of the training log, I guess due to inf loss.
cv_loss is 73 after very first epoch.

getting similar logs after completing one epoch.
@pengzhendong can you please suggest, what parameters need to be tuned?
